### PR TITLE
Adds a new controlledBySW method

### LINKED ISSUE #53: Add a way to "waitForController()" instead of "activateSW()"
This is based off of feedback as a result of skipWaiting and claim in: https://github.com/GoogleChrome/sw-helpers/issues/242

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @gauntface 

Fixes #53

### CODE REVIEW

File: test/browser-tests/window-utils/serviceworkers/immediate-control.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+self.addEventListener('install', () => self.skipWaiting());
```
Comment (gauntface) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: Could you add a comment here just along the lines of:

"This service worker does not use event.waitUntil() on purpose. This has ambiguity over implementation across browsers (i.e. whether it's needed to be used with skipWaiting and clients.claim()). This service worker is used to test controlledBySw() - PLEASE DO NOT CHANGE."

Without this I can see someone eventually sneaking in event.waitUntil() (i.e. myself) and then it's undermines the purpose of this new method.

Comment (jeffposnick) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: I'm not sure what the ambiguity is, though? AFAIK there are no browsers that ever required an `event.waitUntil()` wrapper around `self.skipWaiting()` or `self.clients.claim()`.

There are old examples that used `event.waitUntil()` (including many I wrote), but it turns out it was never actually necessary.

Comment (jeffposnick) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: (I added a comment anyway!)

Comment (gauntface) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: But we've seen the difference in behavior - so it's not "required" it does affect the end behavior.

Comment (jeffposnick) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: PTAL at the revised comment.

Comment (gauntface) [inline test/browser-tests/window-utils/serviceworkers/immediate-control.js]: yeah that works.

**Comment by jeffposnick** [comment]
I've made `eslint` happier with some of the latest commits, but the actual tests are failing now with

```sh
WebDriverError: Unable to parse new session response: {"value": {"sessionId":"e8da8745-f4ac-46c5-ba1f-de5b36a55a4d","value":{"acceptInsecureCerts":false,"browserName":"firefox","browserVersion":"55.0a1","moz:accessibilityChecks":false,"moz:processID":7046,"moz:profile":"/tmp/rust_mozprofile.RHnTV5h5iha9","pageLoadStrategy":"normal","platformName":"linux","platformVersion":"4.8.12-040812-generic","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000}}}}
```

I'm going to add in another commit that bumps all the `devDependencies`and see if that helps, unless you have other ideas, @gauntface.

**Comment by jeffposnick** [comment]
Updating the dependencies is pretty involved, unfortunately, so I'll tackle it in a separate PR that will need to be merged first.

**Comment by jeffposnick** [comment]
This PR is blocked on https://github.com/GoogleChrome/sw-testing-helpers/issues/55, as I haven't had any luck getting the test suite to run properly in Firefox 😦

**Comment by gauntface** [comment]
@jeffposnick if you can switch back to let instead of disabling the lint check, I'll merge the PR and will tackle the tests in a follow up PR if you like.

**Comment by jeffposnick** [comment]
Oops—my forked copy was out of date with the upstream master, and I didn't realize that the rest of that file had switched from `var` to `let`. I couldn't figure out why linting was unhappy with just that new line... it's switched to `let` now.

**Comment by gauntface** [comment]
@jeffposnick Tank you :) Will make sure lint is at least happy + Chrome and then will merge. Sorry for the faff.

**Pull request merged**